### PR TITLE
Attempt to fix mypy failure on ci

### PR DESCRIPTION
**Pull request by vfdev-5**
Fixed failing job "Lint / python-types / linux-job" : https://github.com/pytorch/vision/actions/runs/6955317152/job/18923844284?pr=8128

### CODE REVIEW

File: torchvision/datasets/vision.py [hunk 1]
```diff
@@ -37,7 +37,11 @@ def __init__(
         _log_api_usage_once(self)
         if isinstance(root, str):
             root = os.path.expanduser(root)
-        self.root = root
+        # This lines to please mypy
+        if root is not None:
+            self.root: str = root
+        else:
+            self.root = root  # type: ignore[assignment]
```
Comment (NicolasHug) [inline torchvision/datasets/vision.py:44]: Thanks for fixing this @vfdev-5 

Would it work to just write this line unconditionally?

If there's no cleaner why to please mypy here, I'd suggest to stop type-checking this part of the code-base. It's not worth wasting time on.

Comment (vfdev-5) [reply torchvision/datasets/vision.py:44]: I'm not sure if we can write it in single line...

Technically, mypy is right as `root` arg is `Optional[str]`, so `self.root` type is `Optional[str]`. In subclasses we are passing root string, but `self.root` inherits the original type: `Optional[str]`.

we can force the cast as (+/- some type ignores)
```
self.root: str = root
```
but this wont be correct.

Alternatively, as you say we can stop type checking this code and all related subclasses...

Comment (vfdev-5) [reply torchvision/datasets/vision.py:44]: Just seen that this PR is on the origin of this problem: https://github.com/pytorch/vision/pull/8124

**Comment by pmeier** [comment]
Superseded by #8134.

**Pull request closed without merging**